### ISSUE TITLE
[streamanalytics] Decoding of error JSON response fails

### ISSUE THREAD

**Issue by thsutton** [Azure/azure-sdk-for-go#942]
Attempting to `CreateOrUpdate` a job that is already running can get API responses like this:

```
HTTP/1.1 409 Conflict
....

{"code":"Conflict","message":"The Stream Analytics job is not in a valid state to perform this operation.","details":{"code":"409","message":"The Stream Analytics job is not in a valid state to perform this operation.","correlationId":"uuuuuuuu-iiii-iiii-iiii-dddddddddddd","requestId":"uuuuuuuu-iiii-iiii-iiii-dddddddddddd"}
```

But the SDK eats this helpful information and instead returns an error like:

```
streamanalytics.StreamingJobsClient#CreateOrReplace: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="Unknown" Message="Unknown service error"
```

This happens because `(client StreamingJobsClient).CreateOrReplaceSender` uses the [`azure.WithErrorUnlessStatusCode`](https://github.com/Azure/go-autorest/blob/master/autorest/azure/azure.go#L183) decorator which fails to unmarshal error response JSON with the following error:

```
json: cannot unmarshal object into Go struct field ServiceError.details of type []interface {}
```

Hacking the definition of `ServiceError` to ignore the `Details` field results in the error being decoded correctly:

```
type ServiceError struct {
	Code    string         `json:"code"`
	Message string         `json:"message"`
	Details *[]interface{} `json:"-"`
}
```

Unmarshalls the error response correctly:

```
streamanalytics.StreamingJobsClient#CreateOrReplace: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="Conflict" Message="The Stream Analytics job is not in a valid state to perform this operation."
```

What's the best way to fix this mismatch between Stream Analytics' JSON responses having `details` as an object and the SDK's expecting `details` to be a list of them?

Apropos: I'm interested in trying to tackle some of the issues I'm running into myself but have found it hard to find documentation about getting started with autorest, etc. Is there anything you can point me at to get up and running trying to fix some of these issue instead of just dumping them in your issues?

**Comment by joshgav** [comment]
Thanks @thsutton and thanks for your offer to help! Some instruction on getting started with Autorest for Go are in this repo at https://github.com/Azure/azure-sdk-for-go/blob/master/documentation/code-generation.md. @mcardosos and @marstr can point out more.

@jhendrixMSFT wants to improve error handling (sic) too, see https://github.com/Azure/go-autorest/issues/224.

**Comment by joshgav** [comment]
related: https://github.com/Azure/azure-sdk-for-go/issues/891

**Comment by jhendrixMSFT** [comment]
I *think* that Details should be defined as follows.
```golang
Details *map[string]interface{}
```
I'm working on setting up a repro to test this.

**Comment by jhendrixMSFT** [comment]
Changing it to map works but breaks a bunch of tests.  However changing it to be just ```*interface{}``` works as expected; this might be the way to go but I need to investigate further.

**Comment by jhendrixMSFT** [comment]
Hmm looking at the [OData v4 spec](http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091) the Details field is supposed to be an array of JSON objects.

**Comment by jhendrixMSFT** [comment]
Indented error response for easy reading.
```json
{
  "code": "Conflict",
  "message": "The Stream Analytics job is not in a valid state to perform this operation.",
  "details": {
    "code": "409",
    "message": "The Stream Analytics job is not in a valid state to perform this operation.",
    "correlationId": "uuuuuuuu-iiii-iiii-iiii-dddddddddddd",
    "requestId": "uuuuuuuu-iiii-iiii-iiii-dddddddddddd"
  }
```

**Comment by jhendrixMSFT** [comment]
Fixed in v14.0.0.
